src/components/Table/GridTable.tsx

```diff
@@ -292,14 +292,14 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
 
   const columnSizes = useSetupColumnSizes(style, columns, tableRef, resizeTarget);
 
+  // Make a single copy of our current collapsed state, so we'll have a single observer.
+  const collapsedIds = useComputed(() => rowState.collapsedIds, [rowState]);
```

Comment:
The wonkiness fix is moving this to it's own `useComputed`, and moving the other `useComputed` back to a `useMemo`.

The issue was that, just the way `useComputed` works today, when mobx detects its value is dirty, i.e. it's observer is invoked, `useComputed`'s lambda is immediately run ... where ever that observer was triggered, i.e. in some browser "onclick" callback for collapsing the rows.

Usually that's fine, but the current `useComputed`-now-back-to-`useMemo` does quite a bit of stuff, in particular instantiation of JSX `<GridRow... />` lines that expect to only be done within the context of a React render pass so that things like context propagation works properly.

In particular, the wonkiness was that the `useComputed` as before would actually get invoked, but the updated state, i.e. the new filtered rows, wouldn't actually make their way into the DOM, so it looked like the click got dropped.